tests/deal.II/nedelec_non_rect_face.cc

```diff
@@ -0,0 +1,453 @@
+                cell_rhs(i) += rhs_value_vector*fe_values[E_re].value(i,q_point)*fe_values.JxW(q_point);
+
+              }
+          }
+
+        cell->get_dof_indices (local_dof_indices);
+        constraints.distribute_local_to_global(cell_matrix,
+                                               cell_rhs,
+                                               local_dof_indices,
+                                               system_matrix, system_rhs);
+      }
+  }
+  template <int dim>
+  void MaxwellProblem<dim>::solve ()
+  {
+    /* Direct */
+    SparseDirectUMFPACK A_direct;
+    A_direct.initialize(system_matrix);
+
+    A_direct.vmult (solution, system_rhs);
+    constraints.distribute (solution);
+
+  }
+  template<int dim>
+  void MaxwellProblem<dim>::process_solution(const unsigned int cycle)
+  {
+
+    Vector<double> diff_per_cell(triangulation.n_active_cells());
+
+
+    VectorTools::integrate_difference(mapping, dof_handler, solution, exact_solution,
+                                      diff_per_cell, QGauss<dim>(quad_order+2),
+                                      VectorTools::L2_norm);
+
+
+    const double L2_error = diff_per_cell.l2_norm();
+
+    double Hcurl_error = calcErrorHcurlNorm();
+
+    convergence_table.add_value("cycle", cycle);
+    convergence_table.add_value("cells", triangulation.n_active_cells());
+    convergence_table.add_value("dofs", dof_handler.n_dofs());
+    convergence_table.add_value("L2 Error", L2_error);
+    convergence_table.add_value("H(curl) Error", Hcurl_error);
+
+  }
+
+  template <int dim>
+  void MaxwellProblem<dim>::run ()
+  {
+
+    unsigned int numcycles=3;
+
+    for (unsigned int cycle=0; cycle<numcycles; ++cycle)
+      {
+        if (cycle == 0)
+          {
+            /* Cube mesh */
+            GridGenerator::hyper_cube (triangulation, 0, 1);
+            GridTools::distort_random (0.2, triangulation, false);
+
+
+            /* Testing hanging nodes
+             *        triangulation.refine_global(1);
+             *        cell = triangulation.begin_active();
+             *        cell->set_refine_flag(); // create single refined cell
+             *        triangulation.execute_coarsening_and_refinement();
+             */
+
+          }
+        else
+          {
+            triangulation.refine_global (1);
+          }
+
+        setup_system ();
+        assemble_system ();
+        solve ();
+        process_solution (cycle);
+      }
+
+    convergence_table.set_precision("L2 Error",8);
+    convergence_table.set_scientific("L2 Error",true);
+
+    convergence_table.set_precision("H(curl) Error",8);
+    convergence_table.set_scientific("H(curl) Error",true);
+
+    convergence_table.write_text(std::cout);
+
+  }
+}
+// END MAXWELL CLASS
+int main (int argc, char *argv[])
+{
+  using namespace Maxwell;
+
+  for (unsigned int p=0; p<4; ++p)
+    {
+      std::cout << "p = " << p << std::endl;
+      MaxwellProblem<3> (p).run();
+    }
+
```

Comment:
If you use the same style for main as other tests, you should also get output with less decimals. You should use ctest to generate the outputfile and then copy it to the location for comparison